herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -2043,6 +2061,146 @@ private static final long sumOverflowWise(long a, long b) {
         return total < 0 ? Long.MAX_VALUE : total;
     }
 
+    private static final class CleanAndCompactResult {
+
+        private final DataPage buildingPage;
+        private final boolean keepFlushedPageInMemory;
+        private final List<Long> flushedPages;
+        private final long flushedRecords;
+
+        public CleanAndCompactResult(DataPage buildingPage, boolean keepFlushedPageInMemory, List<Long> flushedPages, long flushedRecords) {
+            super();
+            this.buildingPage = buildingPage;
+            this.keepFlushedPageInMemory = keepFlushedPageInMemory;
+            this.flushedPages = flushedPages;
+            this.flushedRecords = flushedRecords;
+        }
+
+    }
+
+    /**
+     *
+     * @param workingPages            pages to handle for cleanup and compaction
+     * @param buildingPage            currently building page (possibly partially filled)
+     * @param keepFlushedPageInMemory should currently (provided) building page kept on memory or
+     *                                discarded?
+     * @param processLimitInstant     timestamp at which blocking method invocation and return a result
+     *                                (at least a page will be handled)
+     *
+     * @return clean and compact result object
+     */
+    private CleanAndCompactResult cleanAndCompactPages(List<WeightedPage> workingPages, DataPage buildingPage,
+                                                       boolean keepFlushedPageInMemory, long processLimitInstant) {
+
+        long buildingPageSize = buildingPage.getUsedMemory();
+
+        List<Long> flushedPages = new ArrayList<>();
+        long flushedRecords = 0;
+
+        /* Rebuild dirty pages with only records to be kept */
+        for (WeightedPage weighted : workingPages) {
```

Comment:
what about renaming the variable "weighted"  to "page" or "pageToRebuild" or something more meaning full ?

Reply:
renamed just to page